src/main/java/TaskManager.java

```diff
@@ -0,0 +1,90 @@
+import java.util.ArrayList;
+
+public class TaskManager {
+    private ArrayList<Task> taskList;
+
+    public TaskManager() {
+        this.taskList = new ArrayList<Task>();
+    }
+
+    public void parseCommand(String command) throws DukeException {
+        if (command.startsWith("list")) {
+            printList();
+        } else if (command.startsWith("done ")) {
+            String[] temp = command.split(" ");
+            int doneIndex = Integer.parseInt(temp[1]) - 1;
+            markTaskDone(doneIndex);;
+        } else if (command.startsWith("delete ")) {
+            String[] temp = command.split(" ");
+            int delIndex = Integer.parseInt(temp[1]) - 1;
+            deleteTask(delIndex);;
+        } else if (command.startsWith("deadline")) {
+            if (command.length() <= 9){
+                throw new DukeException("Sorry, the description of a deadline cannot be empty!");
+            }
+            addNewDeadline(command.substring(9));
+        } else if (command.startsWith("event")) {
+            if (command.length() <= 6){
+                throw new DukeException("Sorry, the description of a event cannot be empty!");
+            }
+            addNewEvent(command.substring(6));
+        } else if (command.startsWith("todo")) {
+            if (command.length() <= 5){
+                throw new DukeException("The description of a todo cannot be empty!");
```

Comment:
You could probably consider creating custom exceptions.

Reply:
Hmm that has not crossed my mind before. Will consider it for the next version.